subprojects/ide/src/main/java/org/gradle/plugins/ide/internal/resolver/IdeDependencySet.java

```diff
@@ -0,0 +1,233 @@
+        private final Table<ModuleComponentIdentifier, Class<? extends Artifact>, Set<ResolvedArtifactResult>> auxiliaryArtifacts = HashBasedTable.create();
+
+        public void visit(IdeDependencyVisitor visitor) {
+            resolvePlusConfigurations(visitor);
+            resolveMinusConfigurations(visitor);
+            resolveAuxiliaryArtifacts(visitor);
+            visitArtifacts(visitor);
+            visitUnresolvedDependencies(visitor);
+        }
+
+        private void resolvePlusConfigurations(IdeDependencyVisitor visitor) {
+            for (Configuration configuration : plusConfigurations) {
+                ArtifactCollection artifacts = getResolvedArtifacts(configuration, visitor);
+                for (ResolvedArtifactResult resolvedArtifact : artifacts) {
+                    resolvedArtifacts.put(resolvedArtifact.getId(), resolvedArtifact);
+                }
+                if (artifacts.getFailures().isEmpty()) {
+                    continue;
+                }
+                for (UnresolvedDependencyResult unresolvedDependency : getUnresolvedDependencies(configuration, visitor)) {
+                    unresolvedDependencies.put(unresolvedDependency.getAttempted(), unresolvedDependency);
+                }
+            }
+        }
+
+        private void resolveMinusConfigurations(IdeDependencyVisitor visitor) {
+            for (Configuration configuration : minusConfigurations) {
+                ArtifactCollection artifacts = getResolvedArtifacts(configuration, visitor);
+                for (ResolvedArtifactResult resolvedArtifact : artifacts) {
+                    resolvedArtifacts.remove(resolvedArtifact.getId());
+                }
+                if (artifacts.getFailures().isEmpty()) {
+                    continue;
+                }
+                for (UnresolvedDependencyResult unresolvedDependency : getUnresolvedDependencies(configuration, visitor)) {
+                    unresolvedDependencies.remove(unresolvedDependency.getAttempted());
+                }
+            }
+        }
+
+        private ArtifactCollection getResolvedArtifacts(Configuration configuration, final IdeDependencyVisitor visitor) {
+            return configuration.getIncoming().artifactView(new Action<ArtifactView.ViewConfiguration>() {
+                @Override
+                public void execute(ArtifactView.ViewConfiguration viewConfiguration) {
+                    viewConfiguration.lenient(true);
+                    viewConfiguration.componentFilter(getComponentFilter(visitor));
+                }
+            }).getArtifacts();
+        }
+
+        private Spec<ComponentIdentifier> getComponentFilter(IdeDependencyVisitor visitor) {
+            return visitor.isOffline() ? NOT_A_MODULE : Specs.<ComponentIdentifier>satisfyAll();
+        }
+
+        private Iterable<UnresolvedDependencyResult> getUnresolvedDependencies(Configuration configuration, IdeDependencyVisitor visitor) {
```

Comment:
Maybe move/add a comment to here about using `ResolutionResult` due to inadequate failure reporting.